### ISSUE TITLE
Progressive Val Score - Metric For Logistic Regression

### ISSUE THREAD

**Issue by SuleymanCetiner** [online-ml/river#1050]
Hello there,

I have an issue for running just a basic script : 

from river import linear_model
from river import metrics
from river import evaluate
from river import preprocessing
import pandas as pd

data = pd.read_csv("C:/Users/Monster/Desktop/LveR.csv")

# Import label encoder
from sklearn import preprocessing
from sklearn.model_selection import train_test_split
# label_encoder object knows how to understand word labels.
label_encoder = preprocessing.LabelEncoder()

# Encode labels in column 'species'.
data['Class'] = label_encoder.fit_transform(data['Class'])

data['Class'].unique()

X=data.iloc[:,:-1]
y=data.iloc[:,-1]

X_train,X_test,y_train,y_test = train_test_split(X,y,random_state = 11,test_size=0.25,shuffle=True)

model = (
    preprocessing.StandardScaler() |
    linear_model.LogisticRegression())

metric=metrics.ROCAUC()
evaluate.progressive_val_score(X_test,y_test,model,metric)

## That says " Pipeline object has no attribute "works_with". I get confused. Any type of help will be appreciated. Thanks ! 

![Error ](https://user-images.githubusercontent.com/38746955/193858901-a1c979e3-89d7-45f3-892a-dfea8066ef06.png)

**Comment by VaysseRobin** [comment]
Hello, I think the problem is that you import the `preprocessing` module from river first and then you import the `preprocessing` module from sklearn so you overide the previous one. The `StandardScaler` in your model is thus the one from sklearn while you should use the one from river.

**Comment by SuleymanCetiner** [comment]
Hello,i tried that overriding but error still remains ,i will try another things. Thanks !

**Comment by VaysseRobin** [comment]
Ok, I found the issue. `progressive_val_score` expect a dataset and not an X and y ([here is the documentation](https://riverml.xyz/0.13.0/api/evaluate/progressive-val-score/)) so you should have something like that: 

```python
dataset = [({"feature1": 2, "feature2":4}, 0),
          ({"feature1": 3, "feature2":1}, 1),
           ....]

evaluate.progressive_val_score(dataset, model, metric)
```
You can use the [iter_csv](https://riverml.xyz/0.13.0/api/stream/iter-csv/) or any function from the `stream` module to convert your data to a stream.

**Comment by SuleymanCetiner** [comment]
Yes, it makes sense because in general, it asks just one parameter to pass through pipeline. Thank you for help.
